src/javascript/binary/pages/portfiliows.js

```diff
@@ -0,0 +1,159 @@
+
+        var rowTemplate = '<div class="grd-parent grd-grid-12 grd-row-padding table-body table-body-lines"><div class="grd-grid-2 grd-hide-mobile grd-grid-phablet-2">!ref-number!</div><div class="grd-grid-5 grd-grid-mobile-7 grd-grid-phablet-6">!longcode!</div><div class="grd-grid-5 grd-grid-mobile-5 grd-grid-phablet-4"><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 right-aligned">!currency! <strong>!buy_price!</strong></div><div class="grd-hide grd-show-mobile grd-grid-mobile-12 grd-show-phablet grd-grid-phablet-12"><br>Indicative :</div><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 right-aligned">!currency! 0.00</div><div class="grd-hide grd-show-mobile grd-grid-mobile-12 grd-show-phablet grd-grid-phablet-12"><br></div><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 center-aligned"><div><span class="button"><button class="button open_contract_details" onclick="return false;" contract_id="5266199068" controller_action="sell" currency="!currency!" l="!lang-code!" payout="!payout!" purchase_price="!buy_price!" purchase_time="2015-10-02 08:24:13" qty="1" recomputed_price="0" sell_channel="P_!contract_type!-1443774253_!symbol!_1444424400_14958600_14065600_c_!currency!_!lang-code!" shortcode="!contract_type!_!symbol!_100_1443774253_1444424400F_14958600_14065600" url="https://www.binary.com/trade/analyse_contract?l=!lang-code!">:view:</button></span></div></div></div></div>';
+        var contracts = '';
+        var sumPurchase = 0.0;
+        var sumIndicative = 0.0; // As of writing this, the API still does not return a value for this field
+        var currency;
+        //var n, s;
+        $.each(data.portfolio.contracts, function(ci, c) {
+            sumPurchase += parseFloat(c.buy_price, 10);
+            currency = c.currency;
+            contracts += rowTemplate
+            .split("!ref-number!").join(c.fmb_id)
+            .split("!longcode!").join(c.longcode)
+            .split("!currency!").join(c.currency)
+            .split("!lang-code!").join(pageLanguage)
+            .split("!payout!").join(c.payout)
+            .split("!contract_type!").join(c.contract_type)
+            .split("!symbol!").join(c.symbol.toUpperCase())
+            .split("!buy_price!").join(fixCurrency(c.buy_price));
+        });
+
+        // contracts is ready to be added to the dom
+        $("#portfolio-dynamic").replaceWith(trans(contracts));
+
+        // update footer area data
+        $("#cost-of-open-positions").replaceWith( fixCurrency(sumPurchase, currency));
+        $("#value-of-open-positions").replaceWith(currency + '0.00'); // Fix this once API returns a value for indicative amount
+
+        // ready to show portfolio table
+        $("#portfolio-table").removeClass("dynamic");
+
+    };
+
+
+    /*** utility functions ***/
+
+    // Until we can publish language specific json objects to each gh-pages branch
+    var uiTexts = {
+        "EN": {"view":"View"},
```

Comment:
I don't think this should be done like this. All our language translations should be in the language files, otherwise we'll never remember which places translations are peppered around the code. Please check with Raunak how to refactor this.